WordPress/WordPressStatsWidgets/Remote service/StatsWidgetsService.swift

```diff
@@ -123,10 +124,56 @@ class StatsWidgetsService {
             }
             self.state = .ready
         }
+    }
+
+    private func fetchThisWeekStats(service: StatsServiceRemoteV2,
+                                    widgetData: HomeWidgetThisWeekData,
+                                    completion: @escaping (Result<HomeWidgetData, Error>) -> Void) {
+
+        // Get the current date in the site's time zone.
+        let siteTimeZone = widgetData.timeZone
+        let weekEndingDate = Date().convert(from: siteTimeZone).normalizedDate()
+
+        // Include an extra day. It's needed to get the dailyChange for the last day.
+        service.getData(for: .day, endingOn: weekEndingDate,
+                        limit: ThisWeekWidgetStats.maxDaysToDisplay + 1) { [weak self] (summary: StatsSummaryTimeIntervalData?, error: Error?) in
+
+            guard let self = self else {
+                return
+            }
+
+            if let error = error {
+                DDLogError("This Week Widget: Error fetching summary: \(String(describing: error.localizedDescription))")
+                completion(.failure(error))
+                self.state = .error
+                return
+            }
+
+            let summaryData = summary?.summaryData.reversed() ?? []
+            let newWidgetData = HomeWidgetThisWeekData(siteID: widgetData.siteID,
+                                                       siteName: widgetData.siteName,
+                                                       url: widgetData.url,
+                                                       timeZone: widgetData.timeZone,
+                                                       date: Date(),
+                                                       stats: ThisWeekWidgetStats(days: ThisWeekWidgetStats.daysFrom(summaryData: summaryData)))
+            completion(.success(newWidgetData))
 
+            DispatchQueue.global().async {
```

Comment:
I'm not sure if we've changed this to `main` elsewhere now?

Reply:
same here, we should review this once we merge [this](https://github.com/wordpress-mobile/WordPress-iOS/pull/15838)